zookeeper-server/src/test/java/org/apache/zookeeper/server/persistence/TxnLogToolkitTest.java

```diff
@@ -95,27 +106,33 @@ public void testInitMissingFile() throws FileNotFoundException, TxnLogToolkit.Tx
 
     @Test
     public void testMultiTxnDecode() throws IOException {
-        //MultiTxn with four ops, and the first op error.
+        //MultiTxn with multi ops, the first op is error and the others are normal
         List<Txn> txns = new ArrayList<>();
-        int type = -1;
-        for (int i = 0; i < 4; i++) {
-            ErrorTxn txn;
-            if (i == 0) {
-                txn = new ErrorTxn(KeeperException.Code.NONODE.intValue());
-            } else {
-                txn = new ErrorTxn(KeeperException.Code.RUNTIMEINCONSISTENCY.intValue());
-            }
+        Map<Record, Integer> records = new HashMap<>();
+        records.put(new ErrorTxn(KeeperException.Code.NONODE.intValue()), ZooDefs.OpCode.error);
+        records.put(new ErrorTxn(KeeperException.Code.RUNTIMEINCONSISTENCY.intValue()), -1);
+        records.put(new CreateTxn("/test", "test-data".getBytes(StandardCharsets.UTF_8), ZooDefs.Ids.OPEN_ACL_UNSAFE, true, 1), ZooDefs.OpCode.create);
+        records.put(new CreateContainerTxn("/test_container", "test-data".getBytes(StandardCharsets.UTF_8), ZooDefs.Ids.OPEN_ACL_UNSAFE, 2), ZooDefs.OpCode.createContainer);
+        records.put(new CreateTTLTxn("/test_container", "test-data".getBytes(StandardCharsets.UTF_8), ZooDefs.Ids.OPEN_ACL_UNSAFE, 2, 20), ZooDefs.OpCode.createTTL);
+        records.put(new SetDataTxn("/test_set_data", "test-data".getBytes(StandardCharsets.UTF_8), 4), ZooDefs.OpCode.setData);
+        records.put(new DeleteTxn("/test_delete"), ZooDefs.OpCode.delete);
+        records.put(new CheckVersionTxn("/test_check_version", 5), ZooDefs.OpCode.check);
+
+        for (Map.Entry<Record, Integer> recordEntry : records.entrySet()) {
             try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
                 BinaryOutputArchive boa = BinaryOutputArchive.getArchive(baos);
-                txn.serialize(boa, "request");
+                recordEntry.getKey().serialize(boa, "request");
                 ByteBuffer bb = ByteBuffer.wrap(baos.toByteArray());
-                txns.add(new Txn(type, bb.array()));
+                txns.add(new Txn(recordEntry.getValue(), bb.array()));
             }
         }
```

Comment:
How about adding a helper method, say `newSubTxn(type, record)`, to construct `Txn` ? This way we can add test case using `txns.add` sequentially.